packages/react-native/Libraries/Blob/URL.js

```diff
@@ -61,22 +65,22 @@ function validateBaseUrl(url: string) {
   );
 }
 
+
 export class URL {
   _url: string;
-  _searchParamsInstance: ?URLSearchParams = null;
-
-  static createObjectURL(blob: Blob): string {
-    if (BLOB_URL_PREFIX === null) {
-      throw new Error('Cannot create URL for blob!');
+  _parsedUrl: URL | null = null;
+
+  // Utility to parse the URL once and reuse the parsed object
+  _ensureParsed() {
+    if (!this._parsedUrl) {
+      try {
+        this._parsedUrl = new window.URL(this._url);
+      } catch (error) {
+        throw new Error(Invalid URL: ${this._url});
```

Comment:
What's the purpose of rethrowing an error here, is the exception `new URL` already throws insufficient? https://developer.mozilla.org/en-US/docs/Web/API/URL/URL#exceptions